lib/types/utils.ts

```diff
@@ -1,14 +1,19 @@
-type SysWithId = {
+type Sys = {
   id: string
+  type: string
+  [key: string]: any
 }
 
-export type WithOptionalId<Type extends { sys: unknown }> = Omit<Type, 'sys'> & {
-  sys?: Partial<Omit<Type['sys'], 'id'> & SysWithId>
+type Entity = {
+  sys: Sys
+  [key: string]: any
 }
 
-export type WithId<Type extends { sys: unknown }> = Omit<Type, 'sys'> & {
-  sys: Partial<Omit<Type['sys'], 'id'>> & SysWithId
-}
+type Optional<Type, Keys extends keyof Type> = Partial<Type> & Omit<Type, Keys>
+
+export type WithSysWithAtLeastId = { sys: { id: string; [key: string]: any } }
```

Comment:
Can you have `sys` without `type`?

Reply:
Yes. Or rather, for ContentType creation (the case this was created for) you can specify a type, but it's really useless information we ignore, hence the indexed access